sdk/core/azure-json/src/main/java/com/azure/json/JsonArray.java

```diff
@@ -0,0 +1,377 @@
+     *
+     * @param element a JsonElement object representing one of the valid JSON
+     * types: object, array, string, number, boolean, and null
+     * @return JsonArray object representing the new state of the JsonArray object
+     * after the addition of the new JsonElement object appended.
+     * @throws IllegalArgumentException Thrown when a null JsonElement is passed.
+     * Instantiated JsonElements must be passed. Must pass instantiated JsonNull
+     * to represent JSON null value.
+     */
+    public JsonArray addElement(JsonElement element) throws IllegalArgumentException {
+        nullCheck(element);
+        this.elements.add(element);
+        return this;
+    }
+
+    /**
+     * Adds a JsonElement object as a nested element within the JsonArray object
+     * at a particular index, shifting the existing elements from that index to
+     * the right.
+     *
+     * @param element a JsonElement object representing one of the valid JSON
+     * types: object, array, string, number, boolean, and null
+     * @param index the element index within the JsonArray to add the new JsonElement
+     * object and where to shift from
+     * @return JsonArray object representing the new state of the JsonArray object
+     * after the addition of the new JsonElement object.
+     * @throws IllegalArgumentException Thrown when a null JsonElement is passed.
+     * Instantiated JsonElements must be passed. Must pass instantiated JsonNull
+     * to represent JSON null value.
+     * @throws IndexOutOfBoundsException Thrown when index parameter is < 0 or
+     * > size()
+     */
+    public JsonArray addElement(int index, JsonElement element) throws IllegalArgumentException, IndexOutOfBoundsException {
+        nullCheck(element);
+        this.elements.add(index, element);
+        return this;
+    }
+
+    /**
+     * Sets the index of the JsonArray to the specified JsonElement object,
+     * replacing the existing JsonElement object at this index.
+     *
+     * @param element a JsonElement object representing one of the valid JSON
+     * types: object, array, string, number, boolean, and null
+     * @param index the element index within the JsonArray to replace the current
+     * JsonElement with the newly specified JsonElement object
+     * @return JsonArray object representing the new state of the JsonArray object
+     * after the setting of the new JsonElement object at index.
+     * @throws IllegalArgumentException Thrown when a null JsonElement is passed.
+     * Instantiated JsonElements must be passed. Must pass instantiated JsonNull
+     * to represent JSON null value.
+     * @throws IndexOutOfBoundsException Thrown when index parameter is < 0 or
+     * >= size()
+     */
+    public JsonArray setElement(int index, JsonElement element) throws IllegalArgumentException, IndexOutOfBoundsException {
+        nullCheck(element);
+        this.elements.set(index, element);
+        return this;
+    }
```

Comment:
Do I remember correctly that the outcome of our last meeting was to have this return the removed `JsonElement`?

Reply:
Hi @JonathanGiles just tidying these ones up. Remove does return it (line 144 on), but setElement would return the whole array to allow chaining adds?

Reply:
I'll call @alzimmermsft to weigh in on this.

Reply:
@Robsimpson review Javadoc